content/src/service/snapshots/SnapshotManager.ts

```diff
@@ -140,76 +159,90 @@ export class SnapshotManager {
   }
 
   /** This methods queries the database and builds the snapshots, stores it on the content storage, and saves the metadata */
-  private async generateSnapshot(task?: Database): Promise<void> {
+  private async generateSnapshot(): Promise<void> {
+    // Format the snapshot in a tmp file
+    const tmpFile = path.resolve(this.contentStorageFolder, 'tmp-snapshot-file')
```

Comment:
can you ensure that you will only be processing one file each time?

Reply:
Yes, it can be ensured because `generateSnapshot` is called from only one place inside an awaitable loop